src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLRequest.java

```diff
@@ -7,14 +7,24 @@
 
     private final GraphQLOperationRequest request;
     private final GraphQLResponseProjection responseProjection;
+    private final String customOperationName;
 
     public GraphQLRequest(GraphQLOperationRequest request) {
-        this(request, null);
+        this(request, null, null);
     }
 
     public GraphQLRequest(GraphQLOperationRequest request, GraphQLResponseProjection responseProjection) {
+        this(request, responseProjection, null);
+    }
+
+    public GraphQLRequest(
+        GraphQLOperationRequest request,
+        GraphQLResponseProjection responseProjection,
+        String customOperationName
+    ) {
```

Comment:
IMHO the order of parameters should correspond to the order entities in GraphQL request ("natural" order), meaning that when you construct the query, you first define the name of the operation, then request, and then response projection. So 2 points I would like to change here:
1. Make `String operationName` as a first parameter in constructor(s).
2. Create another constructor `(String operationName, GraphQLOperationRequest request)`.

Open to the descission.

Reply:
Good points. I will make the improvements tomorrow or the day after.